doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,917 @@
+
+We can insert these methods at the end of the ``SpyderPomodoroTimerContainer`` declaration, in the section that our cookiecutter template indicates as ``# --- Public API``.
+
+.. code-block:: python
+
+       def start_pomodoro_timer(self):
+           """Start the timer."""
+           self.pomodoro_timer_status.pause = False
+
+       def pause_pomodoro_timer(self):
+           """Pause the timer."""
+           self.pomodoro_timer_status.pause = True
+
+       def stop_pomodoro_timer(self):
+           """Stop the timer."""
+           self.pomodoro_timer_status.pause = True
+           self.pomodoro_timer_status.countdown = self.pomodoro_timer_status.pomodoro_limit
+
+These methods simply manipulate the ``pause`` field of ``pomodoro_timer_status``, and in the case of ``stop_pomodoro_timer`` the countdown is restarted.
+
+Register the toolbar
+--------------------
+
+A final mandatory step is to go to ``plugin.py`` and register this new toolbar component.
+
+To do this, add ``Plugins.Toolbar`` into our plugin requirements:
+
+.. code-block:: python
+
+       REQUIRES = [Plugins.StatusBar, Plugins.Toolbar]
+
+And connect the toolbar we have created in the container to the spyder toolbar.
+
+.. code-block:: python
+
+           toolbar = self.get_plugin(Plugins.Toolbar)
+           toolbar.add_application_toolbar(container.pomodoro_timer_toolbar)
+
+Review the changes
+------------------
+
+Something that has not been mentioned is that every time we make a change to our code we must restart Spyder so that the plugin is reloaded and we can review the changes.
+
+The first thing we can notice is that we already have the corresponding buttons in the toolbar.
+
+.. image:: images/pd_toolbar_buttons.gif
+   :alt: Pomodoro timer toolbar buttons
+
+The strings that were entered as the ``tip`` parameter in the creation of the actions are shown here as the tooltips of the buttons.
+
+Also, if we check the menu "View > Toolbars", we find that there is a new option under a separator: "Pomodoro Timer Toolbar".
+
+.. image:: images/pd_view_pomodoro_toolbar.png
+   :alt: View > Toolbars menu with "Pomodoro Timer Toolbar" option.
+
+These integrations that streamline the user experience of the components added by the new plugin with the rest of Spyder happen thanks to mixins that are added to Spyder's own Widgets. Consider this when adding custom widgets.
+
+Finally, let's check how the new Pomodoro Timer control buttons in the toolbar interact with the component in the status bar.
+
+.. image:: images/pd_toolbar_statusbar_interact.gif
+   :alt: Interaction between the Pomodoro Timer toolbar and its status bar.
+
+We can notice that there is a small delay of 2 seconds, this is because the time between each timer update is one second. We can improve this response time by explicitly restarting the timer.
```

Comment:
At the end of this section say something like "Now your plugin is ready to publish..." so that it doesn't seem like it was cut.

Reply:
Done